Description: Fix compatibility with Python 3.7 since async has become a reserved word
Author: Josue Ortega <josue@debian.org>
Last-Update: 2018-07-29

--- a/yowsup/demos/cli/layer.py
+++ b/yowsup/demos/cli/layer.py
@@ -559,7 +559,7 @@
             mediaUploader = MediaUploader(jid, self.getOwnJid(), filePath,
                                       resultRequestUploadIqProtocolEntity.getUrl(),
                                       resultRequestUploadIqProtocolEntity.getResumeOffset(),
-                                      successFn, self.onUploadError, self.onUploadProgress, async=False)
+                                      successFn, self.onUploadError, self.onUploadProgress, asyncronous=False)
             mediaUploader.start()
 
     def onRequestUploadError(self, jid, path, errorRequestUploadIqProtocolEntity, requestUploadIqProtocolEntity):
--- a/yowsup/layers/interface/interface.py
+++ b/yowsup/layers/interface/interface.py
@@ -125,7 +125,7 @@
             mediaUploader = MediaUploader(builder.jid, self.getOwnJid(), builder.getFilepath(),
                                       resultRequestUploadIqProtocolEntity.getUrl(),
                                       resultRequestUploadIqProtocolEntity.getResumeOffset(),
-                                      successFn, errorFn, progress, async=True)
+                                      successFn, errorFn, progress, asyncronous=True)
             mediaUploader.start()
 
     def __onRequestUploadError(self, errorEntity, requestUploadEntity, builder, error = None):
--- a/yowsup/layers/protocol_media/mediauploader.py
+++ b/yowsup/layers/protocol_media/mediauploader.py
@@ -9,10 +9,11 @@
 logger = logging.getLogger(__name__)
 
 class MediaUploader(WARequest, threading.Thread):
-    def __init__(self, jid, accountJid, sourcePath, uploadUrl, resumeOffset = 0, successClbk = None, errorClbk = None, progressCallback = None, async = True):
+    def __init__(self, jid, accountJid, sourcePath, uploadUrl, resumeOffset=0, successClbk=None,
+                 errorClbk=None, progressCallback=None, asyncronous=True):
         WARequest.__init__(self)
 
-        self.async = async
+        self.asyncronous = asyncronous
         self.jid = jid
         self.accountJid = accountJid
         self.sourcePath = sourcePath
@@ -30,7 +31,7 @@
         self.sock = socket.socket()
 
     def start(self):
-        if self.async:
+        if self.asyncronous:
             threading.Thread.__init__(self)
             super(MediaUploader, self).start()
         else:
